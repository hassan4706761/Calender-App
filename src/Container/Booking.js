import React, { useState } from "react";
import { Col, Row, Typography } from "antd";
import moment from "moment";
import {
  DownOutlined,
  MinusCircleOutlined,
  PlusCircleOutlined,
} from "@ant-design/icons";
import TImePickerDropdown from "../Components/TImePickerDropdown";

const { Paragraph } = Typography;

const Booking = () => {
  const [pickTime, setPickTime] = useState("00:00");
  const [duration, setDuration] = useState("00:00");
  const [durationInHours, setDurationInHours] = useState(0);
  const [durationInMinutes, setDurationInMinutes] = useState(0);

  function handleTimeChange(time) {
    const timeFormat = moment(time).format("HH:mm");
    setPickTime(timeFormat);
  }

  function handleDurationChange(time) {
    const timing = moment(time).format("HH:mm");
    setDuration(timing);
    const timeSplit = timing.split(":");
    setDurationInHours(Number(timeSplit[0]));
    setDurationInMinutes(Number(timeSplit[1]));
  }

  function handleIncreaseDuration() {
    if (durationInMinutes < 59 && durationInMinutes >= 0) {
      setDurationInMinutes(durationInMinutes + 1);
    } else if (durationInMinutes === 59) {
      setDurationInHours(durationInHours + 1);
      setDurationInMinutes(0);
    }
  }

  function handleDecreaseDuration() {
    if (durationInMinutes > 0) {
      setDurationInMinutes(durationInMinutes - 1);
    }
  }
  return (
    <div>
      <Row>
        <Col span={24}>
          <Paragraph className="headingShow" style={{ userSelect: "none" }}>
            Time
          </Paragraph>
          <Row className="timePicker">
            <TImePickerDropdown
              defaultValue={moment(pickTime, "HH:mm")}
              value={moment(pickTime, "HH:mm")}
              onChange={(time) => handleTimeChange(time)}
            />
            <DownOutlined style={{ fontSize: "20px", marginTop: "7px" }} />
          </Row>
        </Col>
      </Row>
      <Row>
        <Col span={24}>
          <Paragraph className="headingShow" style={{ userSelect: "none" }}>
            Duration
          </Paragraph>
          <Row className="timePicker">
            <PlusCircleOutlined
              onClick={handleIncreaseDuration}
              style={{ fontSize: "24px" }}
            />
            <TImePickerDropdown
              defaultValue={moment(duration, "HH:mm")}
              value={moment(`${durationInHours}:${durationInMinutes}`, "HH:mm")}
              format={"HH:mm"}
              onChange={(time) => handleDurationChange(time)}
            />
            <MinusCircleOutlined
              onClick={handleDecreaseDuration}
              style={{ fontSize: "24px" }}
            />
          </Row>
        </Col>
      </Row>
    </div>
  );
};
export default Booking;
