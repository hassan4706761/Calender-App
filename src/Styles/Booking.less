@textAlign: center;
@datePadding: 15px;

.dateHeadingStyle {
  float: left;
  font-size: 24px;
  text-align: @textAlign;
  color: black;
  padding: @datePadding;
}
.dateHideShowStyle {
  float: right;
  font-weight: normal;
  color: #1890ff;
  text-align: @textAlign;
  padding: @datePadding;
}

.MainRow {
  margin-top: 8rem;
}
.heading {
  font-weight: bolder;
  .dateHeadingStyle();
}

.dateRight {
  .dateHideShowStyle();
}

