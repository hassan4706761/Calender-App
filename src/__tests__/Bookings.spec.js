import React from "react";
import {shallow} from "enzyme";
import Booking from "../Container/Booking";

describe("Bookings Rendering rendering", () => {
    it("mounts without crashing", () => {
        const wrapper = shallow(<Booking />);
        wrapper.unmount();
    });
    test("check dom tree rendering", () => {
        let tree = shallow(<Booking />);
        expect(tree).toMatchSnapshot();
    });
});