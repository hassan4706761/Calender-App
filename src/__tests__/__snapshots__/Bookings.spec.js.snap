// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Suite My Test Case 1`] = `ShallowWrapper {}`;

exports[`date selector rendering check dom tree rendering 1`] = `ShallowWrapper {}`;
